### ISSUE TITLE
CRITICAL: Lost very important recording because of missing autosave

### ISSUE THREAD

**Issue by 8zgs4gt** [SimpleMobileTools/Simple-Voice-Recorder#51]
I am a human rights activist in a country with broken laws. I am using this app from the F-Droid store to record my human rights protests while the smartphone screen is locked and protected with a long password.
I need this app to save the recorded audio automaticly every few minutes. When the police get me, the first thing that they do is to turn off the phone by long holding the power button, remove the simcard and write down the IMEI of the phone.
So when the voice recording is running and the battery is been removed or on phones with non removable battery the power button is been pressed long time, i loose the COMPLETE recording.
This have happened last week. I was been hold down in the police station while they took my phone and turned it off. The voice recording of police officers hitting people and much other things are now lost because the recording was running for about 2 hours and when they shut down the smartphone and removed the simcard the complete recording was deleted.

Its critical for this usecase to have autosave functionality. PLEASE add the autosave functionality to this app.

**Comment by tibbi** [comment]
you might want to use some app that sends the recording periodically to the cloud too in such cases

**Comment by 8zgs4gt** [comment]
Yes, this would help in the case when they start smashing the smartphone. But sadly i was not able to find such app in F-Droid. This recorder was the best thing i was able to find.

Could you please add the autosave-functionality? The next human rights actions start in near future and i would like to avoid loosing a recording as long as the phone is not beeing completely broken.
Autosave would also help when they smash the screen but the mainboard is still functional. Then i can myself only turn off the phone by long pressing the button or waiting until the battery dies. In both cases i would loose a important recording.

**Comment by tibbi** [comment]
it might not be added anytine soon, you might want to use more mature apps for such important moments

**Comment by 8zgs4gt** [comment]
To not let you think i forget that or dont need it any more - i am still waiting patient for this functionality.

**Comment by 8zgs4gt** [comment]
And still waiting. Is this complicated? Autosaving a ongoing recording to not loose it when the battery gets low and phone turns off or other things happen should not be that hard to implement i hope.

**Comment by tibbi** [comment]
yes, it is complicated

**Comment by omnyxikron** [comment]
I recall in some previous version, it *did* operate this way, in that it constantly saved and added to a file (which was accessible to the user) on the go. While a recording was being made, if you refreshed the file listing screen (like by going to a different app and then coming back in), you could even see the file for the recording that was currently being made. And so even if the device lost power or got shut off, the part of the recording that had already made had already been saved and was accessible.

I'm speculating(?) the reason why this change happened is that a different low level Android started being used which didn't allow this real-time save functionality?

**Comment by tusch001** [comment]
@8zgs4gt
@omnyxikron is correct, this is still possible. The only change you have to make is:
Choose mp3 as file type in the app settings (instead of m4a). I also recommend to choose a folder on the internal sd card - also in the settings. Since I'm on an older Android version I don't know if you can still choose a folder this way in Android 10+.
If not - you need a rooted phone to access the data folder.

With the mp3 file type the audio file is still usable even if the phone is forcefully turned off.

If you want to sync it constantly to a server: Some years ago I used an app called folder sync for this - dk.tacit.android.foldersync.pro (don't know if you need the pro version, maybe you can try out the free version)

Try it out - and good luck
